/********************************************************************************
*  This file is part of PEMesh                                                  *
*  Copyright(C) 2022: Daniela Cabiddu                                           *

*                                                                               *
*  Author(s):                                                                   *
*                                                                               *
*     Daniela Cabiddu (daniela.cabiddu@cnr.it)                                  *
*                                                                               *
*     Italian National Research Council (CNR)                                   *
*     Institute for Applied Mathematics and Information Technologies (IMATI)    *
*     Via de Marini, 6                                                          *
*     16149 Genoa,                                                              *
*     Italy                                                                     *
*                                                                               *
*                                                                               *
*                                                                               *
*  This program is free software: you can redistribute it and/or modify it      *
*  under the terms of the GNU General Public License as published by the        *
*  Free Software Foundation, either version 3 of the License, or (at your       *
*  option) any later version.                                                   *
*                                                                               *
*  This program is distributed in the hope that it will be useful, but          *
*  WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY   *
*  or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for  *
*  more details.                                                                *
*                                                                               *
*  You should have received a copy of the GNU General Public License along      *
*  with this program. If not, see <https://www.gnu.org/licenses/>.              *
*                                                                               *
*********************************************************************************/

#ifndef ADDPOINTSDIALOG_H
#define ADDPOINTSDIALOG_H

#include <QDialog>

namespace Ui {
class AddPointsDialog;
}

class AddPointsDialog : public QDialog {
  Q_OBJECT

public:
  explicit AddPointsDialog(QWidget *parent = nullptr);
  ~AddPointsDialog();

  int getSampleIndex() const;
  std::string getSampleName() const;
  int getNumMeshes() const;
  int getNumPoints() const;

private slots:

  void update_img(int index);

private:
  Ui::AddPointsDialog *ui;
};

#endif // ADDPOINTSDIALOG_H
